session/session.go

```diff
@@ -1256,13 +1268,37 @@ func (s *session) SetProcessInfo(sql string, t time.Time, command byte, maxExecu
 	s.processInfo.Store(&pi)
 }
 
-func (s *session) ExecuteInternal(ctx context.Context, sql string) (recordSets []sqlexec.RecordSet, err error) {
+func (s *session) ExecuteInternal(ctx context.Context, sql string, args ...interface{}) (recordSets []sqlexec.RecordSet, err error) {
 	origin := s.sessionVars.InRestrictedSQL
 	s.sessionVars.InRestrictedSQL = true
 	defer func() {
 		s.sessionVars.InRestrictedSQL = origin
 	}()
-	return s.Execute(ctx, sql)
+
+	if span := opentracing.SpanFromContext(ctx); span != nil && span.Tracer() != nil {
+		span1 := span.Tracer().StartSpan("session.ExecuteInternal", opentracing.ChildOf(span.Context()))
+		defer span1.Finish()
+		ctx = opentracing.ContextWithSpan(ctx, span1)
+		logutil.Eventf(ctx, "execute: %s", sql)
+	}
+
+	stmtNodes, err := s.ParseWithParams(ctx, sql, args...)
+	if err != nil {
+		return nil, err
+	}
+	if len(stmtNodes) != 1 {
+		return nil, errors.New("ExecuteInternal() API doesn't support multiple statements")
```

Comment:
Please improve this error message by avoiding exposing the name of the internal function to something like:

"Executing multiple statements internally is not supported".

Reply:
done.